language/move-vm/types/src/values/values_impl.rs

```diff
@@ -191,11 +191,21 @@ pub struct Struct {
 ///   enforces mutability restrictions.
 /// * `move_from` is represented through an owned `GlobalValue`.
 #[derive(Debug)]
-pub struct GlobalValue {
-    status: Rc<RefCell<GlobalDataStatus>>,
-    fields: Rc<RefCell<Vec<ValueImpl>>>,
+enum GlobalValueImpl {
+    None,
+    Fresh {
+        fields: Rc<RefCell<Vec<ValueImpl>>>,
+    },
+    Cached {
+        fields: Rc<RefCell<Vec<ValueImpl>>>,
+        status: Rc<RefCell<GlobalDataStatus>>,
+    },
+    Deleted,
 }
 
+#[derive(Debug)]
+pub struct GlobalValue(GlobalValueImpl);
```

Comment:
The `GlobalValue` is no longer an owned thing being passed between the VM and the data cache. It's more like a storage slot right now. Shall we change its name to something more descriptive? I thought about "GlobalSlot" but that sounds weird and unclear by its own...